File: src/ipc/discovery.ts
import { invoke } from "@tauri-apps/api/core";

export interface DiscoveredDevice {
  ip: string;
  mac: string | null;
  vendor: string | null;
  ssh_open: boolean;
  latency_ms: number | null;
}

export const getDefaultCidr = (): Promise<string | null> => invoke("discovery_default_cidr");
export const scan = (cidr: string): Promise<DiscoveredDevice[]> => invoke("discovery_scan", { cidr });

File: src-tauri/src/commands/discovery.rs
use crate::core::discovery::{self, DiscoveredDevice};

#[tauri::command]
pub fn discovery_default_cidr() -> Option<String> {
    let cidr = discovery::default_cidr();
    log::info!("discovery_default_cidr -> {cidr:?}");
    cidr
}

#[tauri::command]
pub async fn discovery_scan(cidr: String) -> Result<Vec<DiscoveredDevice>, String> {
    discovery::scan(&cidr).await.map_err(|e| e.to_string())
}
